src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -5777,8 +5768,8 @@ void ClassFileParser::post_process_parsed_stream(const ClassFileStream* const st
                        "java.lang.Object cannot implement an interface in class file %s",
                        CHECK);
   }
```

Comment:
This is somewhat strange logic.  Seems like there should be an assert that _super_klass == nullptr then an 'else' at 5779 and an assert that _super_class_index != 0.  The trailing else is confusing and it checks something here at 5775.